cmd/selfupdate/selfupdate.go

```diff
@@ -0,0 +1,327 @@
+package selfupdate
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/cmd"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/config/flags"
+	"github.com/spf13/cobra"
+
+	versionCmd "github.com/rclone/rclone/cmd/version"
+)
+
+// Options contains options for the self-update command
+type Options struct {
+	Check   bool
+	Output  string
+	Beta    bool
+	Stable  bool
+	Version string
+}
+
+// Opt is options set via command line
+var Opt = Options{}
+
+func init() {
+	cmd.Root.AddCommand(cmdSelfUpdate)
+	cmdFlags := cmdSelfUpdate.Flags()
+	flags.BoolVarP(cmdFlags, &Opt.Check, "check", "", Opt.Check, "Check for latest release, do not download.")
+	flags.StringVarP(cmdFlags, &Opt.Output, "output", "", Opt.Output, "Save the downloaded binary at a given path (default: replace running binary)")
+	flags.BoolVarP(cmdFlags, &Opt.Stable, "stable", "", Opt.Stable, "Install latest stable release (this is the default)")
+	flags.BoolVarP(cmdFlags, &Opt.Beta, "beta", "", Opt.Beta, "Install latest beta release.")
+	flags.StringVarP(cmdFlags, &Opt.Version, "version", "", Opt.Version, "Install the given rclone path (default: auto-detect)")
+}
+
+var cmdSelfUpdate = &cobra.Command{
+	Use:     "selfupdate",
+	Aliases: []string{"self-update"},
+	Short:   `Update the rclone binary.`,
+	Long: `
+This command downloads the latest release of rclone and replaces
+the currently running binary. The download is verified with a hashsum.
+
+If you previously installed rclone via a package manager, the package may
+include local documentation or configure services. This command will update
+only rclone executable so the local manual may become inaccurate after it.
```

Comment:
> I note also that if people have installed via a package, maybe we should update the package? Something for the next revision?

This is not the first or second time I see this idea.
I thought about installing .deb/.rpm
TL;DR I wouldn't plan it even for next or next after next revision. :-)

Cons:
1. detection of whether **current binary** was installed directly, via .deb/.rpm package or via install.sh 
     even if one of binaries on the host was installed by a package sometime ago
     any such detection will be fragile
2. to detect .deb/.rpm, we must at least know official rclone path in .deb/.rpm, and keep it in sync with the check in this function
3. to decide whether we want .deb or .rpm for current system, we have to detect the system
     (look for example at the ansible code that does the detection, you'll see a lot of surprises)
     tl;dr it is fragile
4. unlike binary install, package install will **always** require root
5. it brings in yet another external dependency

Pros:
1. it will install manual page

What else is in .deb/.rpm, besides executable?
Custom rclone systemd services? No
Package contains executable, manual page, and MIT license
Who reads manual page when there is rclone.org and rclone --help?
Well, we address rpm/deb. What about .apk, flatpak, snap? We fallback to executable-only update.

Reply:
I was just thinking that we supply .deb and .rpm packages so if the user has installed from a .deb or .rpm we should offer to install our version of those instead.

The question - is this binary installed by a package isn't trivial to answer and I think we should side-step it. If users want to install a deb package then they should do `rclone self-update --deb` which is much simpler to code. This should be limited to downloading the correct `.deb` from the rclone page and running `dpkg -i package.deb` (possibly prefixed by sudo) on it.

I think updating the executable in place like you've done is exactly the way `restic self-update` works and will help enormously with support. Its also the correct way for macOS and Windows - the `deb`, `rpm` thing is of secondary importance I think.

Reply:
done